athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneRecordHandler.java

```diff
@@ -0,0 +1,248 @@
+import com.amazonaws.athena.connector.lambda.domain.predicate.Range;
+import com.amazonaws.athena.connector.lambda.domain.predicate.SortedRangeSet;
+import com.amazonaws.athena.connector.lambda.domain.predicate.ValueSet;
+import com.amazonaws.athena.connector.lambda.handlers.RecordHandler;
+import com.amazonaws.athena.connector.lambda.records.ReadRecordsRequest;
+import com.amazonaws.services.athena.AmazonAthena;
+import com.amazonaws.services.athena.AmazonAthenaClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerClientBuilder;
+
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+//import org.apache.tinkerpop.gremlin.structure.Graph;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+//import org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerGraph;
+import org.apache.tinkerpop.gremlin.driver.Client;
+import org.apache.tinkerpop.gremlin.groovy.engine.GremlinExecutor;
+import org.apache.tinkerpop.gremlin.jsr223.ConcurrentBindings;
+import org.apache.arrow.util.VisibleForTesting;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.Field;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicLong;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * This class is part of an tutorial that will walk you through how to build a
+ * connector for your custom data source. The README for this module
+ * (athena-neptune) will guide you through preparing your development
+ * environment, modifying this example RecordHandler, building, deploying, and
+ * then using your new source in an Athena query.
+ * <p>
+ * More specifically, this class is responsible for providing Athena with actual
+ * rows level data from your source. Athena will call readWithConstraint(...) on
+ * this class for each 'Split' you generated in NeptuneMetadataHandler.
+ * <p>
+ * For more examples, please see the other connectors in this repository (e.g.
+ * athena-cloudwatch, athena-docdb, etc...)
+ */
+public class NeptuneRecordHandler extends RecordHandler {
+    private static final Logger logger = LoggerFactory.getLogger(NeptuneRecordHandler.class);
+
+    /**
+     * used to aid in debugging. Athena will use this name in conjunction with your
+     * catalog id to correlate relevant query errors.
+     */
+    private static final String SOURCE_TYPE = "neptune";
+    private final NeptuneConnection neptuneConnection;
+    private GraphTraversalSource graphTraversalSource;
```

Comment:
can this be final? Also, see my earlier comment about a connection factory to allow for recovering when a connection is bad. I'm not sure how the NeptuneCluster object works internally and if it has auto-recovery already.